receiver/vcenterreceiver/metrics.go

```diff
@@ -460,6 +470,27 @@ func (v *vcenterMetricScraper) recordVMPerformanceMetrics(entityMetric *performa
 			case "net.droppedRx.summation":
 				rxRate := float64(nestedValue) / 20
 				v.mb.RecordVcenterVMNetworkPacketDropRateDataPoint(pcommon.NewTimestampFromTime(si.Timestamp), rxRate, metadata.AttributeThroughputDirectionReceived, val.Instance)
+			case "net.multicastRx.summation":
```

Comment:
Not to change on this PR, but, consider if this `switch` is getting large enough to be better handled by a map of strings to functions. Depending on the number of cases, especially with similar prefixes, it can be that the map gets better performance and smaller code. Anyway, a benchmark should clarify that.

Reply:
This is a good suggestion. The code for the Vcenter has this all over the place. I believe this could be a good first issue, WDYT?

Reply:
Yes, it should be straightforward to implement it. I was just curious to see a benchmark with the perf diff - this is one case that even for readability the change itself is worth, but, I am curious about code size and benchmark.